node/src/bead/mod.rs

```diff
@@ -87,8 +88,47 @@ pub enum BeadResponse {
     Genesis(Vec<BeadHash>),
     // Get all beads for IBD
     GetAllBeads(Vec<Bead>),
+    // Get beads after a specific set of hashes
+    GetBeadsAfter(Vec<BeadHash>),
     // Error response
-    Error(String),
+    Error(BeadSyncError),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum BeadSyncError {
+    GenesisMismatch,
+    Other(String),
+}
+
+impl Encodable for BeadSyncError {
+    fn consensus_encode<W: Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
+        match self {
+            BeadSyncError::GenesisMismatch => 0u8.consensus_encode(w),
```

Comment:
You're hard coding 0u8 and 0u1 as the enum values here. These are magic protocol numbers and should be defined externally. Using the enum as our magic number repository instead, we can do:
```
            BeadSyncError::GenesisMismatch => (*self as u8).consensus_encode(w),
````